docs/notebooks/inequality_constraints.pct.py

```diff
@@ -126,15 +126,15 @@ def create_bo_model(data):
 num_steps = 20
 bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
 
-result = bo.optimize(num_steps, initial_data, models, acquisition_rule=rule)
-
-if result.error is not None: raise result.error
+data = bo.optimize(
+    num_steps, initial_data, models, rule, track_state=False
+).try_get_final_data()
 
 # %% [markdown]
 # To conclude, we visualise the resulting data. Orange dots show the new points queried during optimization. Notice the concentration of these points in regions near the local minima.
 
 # %%
-constraint_data = result.datasets[CONSTRAINT]
+constraint_data = data[CONSTRAINT]
```

Comment:
```suggestion
constraint_data = datasets[CONSTRAINT]
```
as per my comment on expected_improvement notebook ("data" vs "datasets")